libethash/io_posix.c

```diff
@@ -86,9 +90,22 @@ bool ethash_file_size(FILE* f, size_t* ret_size)
 
 bool ethash_get_default_dirname(char* strbuf, size_t buffsize)
 {
+#if defined(__APPLE__)
+	static const char dir_suffix[] = "Ethash/";
+	strbuf[0] = '\0';
+	char* home_dir = getenv("HOME");
+	if (!home_dir || strlen(home_dir) == 0)
```

Comment:
Why are you making this change? If the user has not set a $HOME variable and is in Linux/Mac then it's his problem. The `getpwuid()` function is another way of doing the same thing as querying $HOME. I say we advice our users set $HOME and keep an identical implementation for Posix compliant Operating systems.

If you really want this in I suppose it does not hurt, but then there is absolutely no need for the #ifndef. This will work on all Linux systems too.